File: tp1/py/ej6b/ej6b3c.py
import numpy as np
import matplotlib.pyplot as plt

n = np.arange(-5, 8)

x_n = np.zeros(n.shape)

signal = (n**2 >= 0) & (n**2 <= 4)
x_n[signal] = 1

plt.stem(n, x_n, basefmt=" ")

plt.grid(True)
plt.gca().tick_params(labelsize=18)

plt.show()

File: oct/tp1/ej6b/ej6b3c.m
% tp1 - ej6b - 3c

n = -1:7
x_n = zeros(size(n))
signal = (n >= 0 & n <= 4)
x_n(signal) = 1
stem(n,x_n)

grid on
hold on
set(gca, 'FontSize', 18);

n = -1:7
x_n = zeros(size(m))
signal = (n*2 >= 0 & n*2 <= 4)
x_n(signal) = 1
stem(n,x_n)
